### PR TITLE
Make item list sort setting sticky

### LINKED ISSUE #425: Sort entries option Recent is not kept after closing/opening the app
Lockbox: 1260

Steps.
Having some entries tap on the A-Z button
Select Recently Used option
Once the entries are oreder by Recent, and that option is shown in the navigation bar, close the app
Open the app again

Actual.
The sorted by option changes to A-Z

Expected.
I would expect to have the same option selected when I left the app.

### PR DESCRIPTION
**Pull request by joeyg**
Resolves #425.

### CODE REVIEW

File: lockbox-ios/Presenter/ItemListPresenter.swift [hunk 1]
```diff
@@ -177,13 +181,17 @@ class ItemListPresenter {
          itemListDisplayActionHandler: ItemListDisplayActionHandler = ItemListDisplayActionHandler.shared,
          dataStoreActionHandler: DataStoreActionHandler = DataStoreActionHandler.shared,
          dataStore: DataStore = DataStore.shared,
-         itemListDisplayStore: ItemListDisplayStore = ItemListDisplayStore.shared) {
+         itemListDisplayStore: ItemListDisplayStore = ItemListDisplayStore.shared,
```
Comment (sashei) [inline lockbox-ios/Presenter/ItemListPresenter.swift]: nit: all the other action handlers are declared together.

File: lockbox-ios/Presenter/ItemListPresenter.swift [hunk 2]
```diff
@@ -216,7 +224,7 @@ class ItemListPresenter {
 
         self.view?.bind(sortingButtonTitle: itemSortTextDriver)
 
-        self.itemListDisplayActionHandler.invoke(ItemListSortingAction.alphabetically)
+        self.itemListDisplayActionHandler.invoke(Constant.setting.defaultItemListSort.asAction())
```
Comment (sashei) [inline lockbox-ios/Presenter/ItemListPresenter.swift]: I think this means that the setting will be cleared / changed to the default if you hard-quit the app...

**Comment by joeyg** [comment]
@sashei thanks for the feedback! I just want to make sure I'm following it correctly: I should remove `ItemListSortingAction` and instead have the Presenter (and other classes) just read from `UserDefaults` ?

**Comment by sashei** [comment]
@joeyg It's more about having one source of truth for the setting (in this case `UserDefaults` rather than removing it entirely; I'm happy to keep the `ItemListSortingAction`, similar to the `AutoLockTime` setting, but instead of reading it from the `ItemListDisplayStore`, read it from `UserDefaults`.

Does that make sense? I'm not sure if I'm explaining this well :p

**Comment by joeyg** [comment]
Thanks @sashei   Let me know if the changes I made yesterday are closer to what you are looking for

**Comment by sashei** [comment]
@joeyg apologies, just went back through the code changes. This looks great!!

**Comment by joeyg** [comment]
@sashei this is ready for re-review now that the tests pass.

**Comment by joeyg** [comment]
@devinreams I think this one is ready to go.

**Comment by devinreams** [comment]
Thanks @joeyg 💖

**Pull request merged**